backend/src/migrations/database_optimization.py

```
@@ -0,0 +1,548 @@
+                        EXECUTE format('CREATE TABLE IF NOT EXISTS %I PARTITION OF processing_history_partitioned
+                                       FOR VALUES FROM (%L) TO (%L)',
+                                       partition_name, start_date, end_date);
+                    END LOOP;
+                END $$;
+            """)
+
+            self.db.execute(partition_processing_history)
+            results['processing_history_partitioning'] = 'completed'
+
+            logger.info("Table partitioning completed")
+            return results
```

Ruff (0.14.0)

Consider moving this statement to an `else` block

(TRY300)